src/etc/di.xml

```diff
@@ -8,4 +8,13 @@
     <type name="Magento\Sales\Model\Service\OrderService">
         <plugin name="inventory_sales_source_deduction_processor" type="Ampersand\DisableStockReservation\Plugin\SourceDeductionProcessor"/>
     </type>
+    <!--
+        Fix M2.4 introduced bug caused by new interface implementation.
+        - @see \Magento\InventoryInStorePickupSales\Model\SourceSelection\GetSourceItemQtyAvailableService
+        - @see https://github.com/AmpersandHQ/magento2-disable-stock-reservation/issues/38
+     -->
+    <preference
+        for="Magento\InventorySourceSelectionApi\Model\GetSourceItemQtyAvailableInterface"
+        type="Magento\InventorySourceSelectionApi\Model\GetSourceItemQtyAvailableService"
```

Comment:
Can you describe in a little more detail what this change is please?

Reply:
@convenient As part of M2.4, a new class is set for this interface (`\Magento\InventoryInStorePickupSales\Model\SourceSelection\GetSourceItemQtyAvailableService`). The execute method for this class takes into account in-store reserved quantity:

```php
    public function execute(SourceItemInterface $sourceItem): float
    {
        /* TODO: create config and check if store pickup is enabled? */
        return $sourceItem->getQuantity() - $this->getStorePickupReservedQty($sourceItem);
    }
```

This causes an error with this module as we are disabling the stock reservation functionality - essentially making that negation reduant. Fix here is to revert to the previous implementation for this interface which just returns the source item quantity. Original class execute method: 

```php
    public function execute(SourceItemInterface $sourceItem): float
    {
        return $sourceItem->getQuantity();
    }
```

Reply:
Amazing. 

If you could add a few permalinks to the magento2 repository classes in your comment that would help keep the intention of this clear in future :)